### ISSUE TITLE
Bug with REST API executions-current

### ISSUE THREAD

**Issue by barn4k** [n8n-io/n8n#3268]
**Describe the bug**
When I'm trying to execute an HTTP node with url: `http://localhost:5678/rest/executions-current`, I'm getting error:
```
{
  "status": "rejected",
  "reason": {
    "message": "500 - {\"code\":0,\"message\":\"Cannot read properties of undefined (reading 'toString')\",\"hint\":\"\",\"stack\":\"TypeError: Cannot read properties of undefined (reading 'toString')\\n    at /usr/local/lib/node_modules/n8n/dist/src/Server.js:1533:65\\n    at async /usr/local/lib/node_modules/n8n/dist/src/ResponseHelper.js:90:26\"}",
    "name": "Error",
    "stack": "Error: Request failed with status code 500\n    at createError (/usr/local/lib/node_modules/n8n/node_modules/axios/lib/core/createError.js:16:15)\n    at settle (/usr/local/lib/node_modules/n8n/node_modules/axios/lib/core/settle.js:17:12)\n    at IncomingMessage.handleStreamEnd (/usr/local/lib/node_modules/n8n/node_modules/axios/lib/adapters/http.js:269:11)\n    at IncomingMessage.emit (node:events:539:35)\n    at endReadableNT (node:internal/streams/readable:1345:12)\n    at processTicksAndRejections (node:internal/process/task_queues:83:21)"
  }
}
```

If I enter endpoint in the way of: `http://localhost:5678/rest/executions-current?filter={"workflowId":{{$workflow.id}}, "mode": "trigger"}&limit=10`

I receive another error:
```
{
  "status": "rejected",
  "reason": {
    "message": "500 - {\"code\":0,\"message\":\"Unexpected token , in JSON at position 14\",\"hint\":\"\",\"stack\":\"SyntaxError: Unexpected token , in JSON at position 14\\n    at JSON.parse (<anonymous>)\\n    at /usr/local/lib/node_modules/n8n/dist/src/Server.js:1529:52\\n    at /usr/local/lib/node_modules/n8n/dist/src/ResponseHelper.js:90:32\\n    at Layer.handle [as handle_request] (/usr/local/lib/node_modules/n8n/node_modules/express/lib/router/layer.js:95:5)\\n    at next (/usr/local/lib/node_modules/n8n/node_modules/express/lib/router/route.js:144:13)\\n    at Route.dispatch (/usr/local/lib/node_modules/n8n/node_modules/express/lib/router/route.js:114:3)\\n    at Layer.handle [as handle_request] (/usr/local/lib/node_modules/n8n/node_modules/express/lib/router/layer.js:95:5)\\n    at /usr/local/lib/node_modules/n8n/node_modules/express/lib/router/index.js:284:15\\n    at Function.process_params (/usr/local/lib/node_modules/n8n/node_modules/express/lib/router/index.js:346:12)\\n    at next (/usr/local/lib/node_modules/n8n/node_modules/express/lib/router/index.js:280:10)\\n    at /usr/local/lib/node_modules/n8n/dist/src/UserManagement/routes/index.js:126:9\\n    at Layer.handle [as handle_request] (/usr/local/lib/node_modules/n8n/node_modules/express/lib/router/layer.js:95:5)\\n    at trim_prefix (/usr/local/lib/node_modules/n8n/node_modules/express/lib/router/index.js:328:13)\\n    at /usr/local/lib/node_modules/n8n/node_modules/express/lib/router/index.js:286:9\\n    at Function.process_params (/usr/local/lib/node_modules/n8n/node_modules/express/lib/router/index.js:346:12)\\n    at next (/usr/local/lib/node_modules/n8n/node_modules/express/lib/router/index.js:280:10)\\n    at /usr/local/lib/node_modules/n8n/dist/src/UserManagement/routes/index.js:96:13\\n    at Layer.handle [as handle_request] (/usr/local/lib/node_modules/n8n/node_modules/express/lib/router/layer.js:95:5)\\n    at trim_prefix (/usr/local/lib/node_modules/n8n/node_modules/express/lib/router/index.js:328:13)\\n    at /usr/local/lib/node_modules/n8n/node_modules/express/lib/router/index.js:286:9\\n    at Function.process_params (/usr/local/lib/node_modules/n8n/node_modules/express/lib/router/index.js:346:12)\\n    at next (/usr/local/lib/node_modules/n8n/node_modules/express/lib/router/index.js:280:10)\"}",
    "name": "Error",
    "stack": "Error: Request failed with status code 500\n    at createError (/usr/local/lib/node_modules/n8n/node_modules/axios/lib/core/createError.js:16:15)\n    at settle (/usr/local/lib/node_modules/n8n/node_modules/axios/lib/core/settle.js:17:12)\n    at IncomingMessage.handleStreamEnd (/usr/local/lib/node_modules/n8n/node_modules/axios/lib/adapters/http.js:269:11)\n    at IncomingMessage.emit (node:events:539:35)\n    at endReadableNT (node:internal/streams/readable:1345:12)\n    at processTicksAndRejections (node:internal/process/task_queues:83:21)"
  }
}
```

endpoint **/rest/executions** is working fine

User Management is disabled

Before user Management implementation all these endpoints have worked fine.

**To Reproduce**
1. Copy the workflow:
```
{
  "name": "My workflow 4",
  "nodes": [
    {
      "parameters": {},
      "name": "Start",
      "type": "n8n-nodes-base.start",
      "typeVersion": 1,
      "position": [
        240,
        300
      ]
    },
    {
      "parameters": {
        "url": "=http://localhost:5678/rest/executions-current?filter={\"workflowId\":{{$workflow.id}}, \"mode\": \"trigger\"}&limit=10",
        "options": {}
      },
      "name": "Get executions",
      "type": "n8n-nodes-base.httpRequest",
      "typeVersion": 1,
      "position": [
        460,
        300
      ]
    },
    {
      "parameters": {
        "conditions": {
          "number": [
            {
              "value1": "={{$json[\"data\"].length}}",
              "operation": "larger",
              "value2": 1
            }
          ]
        }
      },
      "name": "IF running",
      "type": "n8n-nodes-base.if",
      "typeVersion": 1,
      "position": [
        620,
        300
      ]
    },
    {
      "parameters": {},
      "name": "Is NOT running - Continue",
      "type": "n8n-nodes-base.noOp",
      "typeVersion": 1,
      "position": [
        800,
        320
      ]
    },
    {
      "parameters": {},
      "name": "Is running - Exit",
      "type": "n8n-nodes-base.noOp",
      "typeVersion": 1,
      "position": [
        800,
        140
      ]
    }
  ],
  "connections": {
    "Get executions": {
      "main": [
        [
          {
            "node": "IF running",
            "type": "main",
            "index": 0
          }
        ]
      ]
    },
    "IF running": {
      "main": [
        [
          {
            "node": "Is running - Exit",
            "type": "main",
            "index": 0
          }
        ],
        [
          {
            "node": "Is NOT running - Continue",
            "type": "main",
            "index": 0
          }
        ]
      ]
    },
    "Start": {
      "main": [
        [
          {
            "node": "Get executions",
            "type": "main",
            "index": 0
          }
        ]
      ]
    }
  },
  "active": false,
  "settings": {}
}
```
2. Run it

**Expected behavior**
/rest/executions-current will output current executions
/rest/executions-current?filter will output current executions with applied filter

**Environment (please complete the following information):**
 - OS: Docker
 - n8n Version: 0.175.1

**Comment by Joffcom** [comment]
Hey @barn4k,

This is possibly one of those things that may not get resolved as we are working on an official public api as our internal api is likely to change and may break.

I will take a look on Monday to see if I can work out what is going on.

**Comment by barn4k** [comment]
If there will be a separate node or another way to get the current workflow executions, it would be great. Main use case is when I have a workflow that triggers in 2 min interval and it's execution time is more than 2 min (for some scenarios).,I don't want to get it executed again while my first execution is still working

**Comment by ivov** [comment]
Are you saving the workflow?

1. `"Unexpected token , in JSON at position 14"` means that `$workflow.id` resolved to nothing, leading to invalid JSON. The only way I can reproduce is with the workflow unsaved.

2. Thanks for flagging the `/executions-current` issue, fixed in the linked PR. Again I could only reproduce this with an unsaved workflow.

If the URL ends with `/workflow` (no ID), then your workflow is unsaved.

**Comment by barn4k** [comment]
@ivov 
Weird.. yes, I didn't save the workflow before trying to get the `/executions-current` endpoint (it was completely unclear for me, that unsaved workflow has no ID). With saved workflow everything seems working fine (even no more auth check with user management disabled).

**Comment by janober** [comment]
Got fixed with #3283
